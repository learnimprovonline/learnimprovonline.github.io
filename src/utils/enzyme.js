import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-15'
import toJson from 'enzyme-to-json'

Enzyme.configure({ adapter: new Adapter() })
