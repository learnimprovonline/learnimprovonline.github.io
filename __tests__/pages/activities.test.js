import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Link from 'gatsby-link'
import ActivitiesPage from '../../src/pages/activities'

const ActivitiesPageDom = shallow(
  <ActivitiesPage />
)

describe('Activities Page', () => {

  describe('Page Heading', () => {
    test('display text is "Activites"', () => {
      const pageTitle = 'Activities'
      const pageHeading = ActivitiesPageDom.find('h1')
      const pageHeadingText = pageHeading.children().text()

      expect(pageHeadingText).toEqual(pageTitle)
    })
  })

  describe('Index Link', () => {
    const indexLink = ActivitiesPageDom.find(Link)

    test('display text is "Go to Home" link', () => {
      const indexLinkText = indexLink.children().text()

      expect(indexLinkText).toEqual('Go to Home')
    })

    test('points to the Index Page', () => {
      const indexPageDestination = '/'
      const indexLinkDestination = indexLink.prop('to')

      expect(indexLinkDestination).toEqual(indexPageDestination)
    })
  })

  test('renders itself', () => {
    const tree = toJson(ActivitiesPageDom)

    expect(tree).toMatchSnapshot()
  })
})





