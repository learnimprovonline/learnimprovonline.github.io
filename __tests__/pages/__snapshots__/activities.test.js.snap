// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activities Page renders itself 1`] = `
<div>
  <h1>
    Activities
  </h1>
  <p>
    Learn improv warm-ups and exercises through a variety of written descriptions, animated instructionals, and video examples. Empowering you and your team the ability to take yourself to the next level.
  </p>
  <GatsbyLink
    to="/"
  >
    Go to Home
  </GatsbyLink>
  <hr />
  <ul>
    <li
      key="0"
    >
      node one
    </li>
    <li
      key="1"
    >
      node two
    </li>
    <li
      key="2"
    >
      node three
    </li>
  </ul>
</div>
`;
